osquery/events/CMakeLists.txt

```diff
@@ -37,8 +47,22 @@ ADD_OSQUERY_LIBRARY(FALSE osquery_additional_events
 file(GLOB OSQUERY_KERNEL_QUEUE_TEST "kernel/tests/*.cpp")
 ADD_OSQUERY_TEST(FALSE ${OSQUERY_KERNEL_QUEUE_TEST})
 
-file(GLOB OSQUERY_EVENTS_TESTS "tests/*.cpp")
-ADD_OSQUERY_TEST(TRUE ${OSQUERY_EVENTS_TESTS})
+# osquery core (additional) set of unit tests built outside of SDK.
+add_executable(osquery_kernel_tests ../main/tests.cpp ${OSQUERY_KERNEL_QUEUE_TEST})
+TARGET_OSQUERY_LINK_WHOLE(osquery_kernel_tests libosquery)
+TARGET_OSQUERY_LINK_WHOLE(osquery_kernel_tests libosquery_additional)
+target_link_libraries(osquery_kernel_tests gtest libosquery_testing)
+SET_OSQUERY_COMPILE(
+  osquery_kernel_tests
+  "${CXX_COMPILE_FLAGS} -DOSQUERY_TEST_DIR='\"osquery-tests-kernel\"' -DGTEST_HAS_TR1_TUPLE=0"
```

Comment:
How does this get set otherwise (in the non-kernel-tests case)? Would it make more sense to drop this define,  remove the const from `kTestWorkingDirectory`, then `+= '-' + std::to_string(getuid());` within main?